src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -575,6 +575,19 @@ public ResponseEntity<EnumerateSnapshotAccessRequest> enumerateSnapshotRequests(
     return ResponseEntity.ok(snapshotBuilderService.enumerateByDatasetId(id));
   }
 
+  @Override
+  public ResponseEntity<SnapshotBuilderGetConceptsResponse> searchConcepts(
+      UUID id, String domainId, String searchText) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    iamService.verifyAuthorization(
+        userRequest,
+        IamResourceType.DATASET,
+        id.toString(),
+        IamAction.VIEW_SNAPSHOT_BUILDER_SETTINGS);
+    return ResponseEntity.ok(
+        snapshotBuilderService.searchConcepts(id, domainId, searchText, userRequest));
+  }
```

Comment:
Yeah I think we could abstract it and we'd need to pass in the IamResourceType and IamAction. I know we are planning to make changes to IamAction so those two actions aren't stretched so thin. This could be a good opportunity to refactor.

Reply:
TDR doesn't currently use spring auth at all, so adding support for annotation based auth checks would be a fair amount of additional work. I think historically the reason for this is because unlike the typical spring usage, we don't do authn inside our app at all; we rely on the proxy to authn user, and then only do authz inside the app. This is true of all all terra spring apps as far as I know.

Once the API layer starts to settle down, we need to generally address authz for the snapshot builder. The usual way to do this is to look at all the endpoints that need auth and create distinct actions in TDR's sam config for each distinct one. Then we create (or extend existing) roles and add actions to the roles. If there isn't a ticket for this work already, we should create one.

I do think there's an opportunity to improve TDR's auth code and generalize it (and also stop using `AuthenticatedUserRequest` which has been deprecated for a while now) but that would be a task separate from the snapshot builder work of adding real authz checks. This would probably start with a (new) spike ticket to look into the best way to do this.